communitymanagement/src/main/java/communitymanagement/controller/TicketController.java

```diff
@@ -100,4 +102,34 @@ public TicketsResident getResidentTickets(@RequestParam(value = "user") int user
 		return ticketsResident;
 	}
 	
+	@PostMapping("/tickets/{ticket_id}/staff-action")
+	public ResponseEntity<String> ticketStatusChangedByStaffAction(
+			@PathVariable(value = "ticket_id") int ticketId, @RequestBody Map<String, String> reqBody) {
+		// Get ticket
+		Ticket ticket = ticketService.getTicketById(ticketId);
+		if (ticket == null) {
+			return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Ticket with id " + ticketId + " not found!");
+		}
+
+		// Change status
+		String action = reqBody.get("action");
+		try {
+			if (action.equals("accept")) {
+				ticketService.updateTicketStatus(ticketId, TicketStatus.INPROGRESS);
+				return ResponseEntity.status(HttpStatus.ACCEPTED).body("Staff accepted: ticket is in progress now");
+			} else if (action.equals("decline")) {
+				ticketService.updateTicketStatus(ticketId, TicketStatus.OPEN);
```

Comment:
we will also need to remove assignment from the ticket_work_assignment table if the staff decline a ticket

Reply:
will be done in another PR